components/properties/src/props.rs

```diff
@@ -416,3 +422,98 @@ impl Script {
     pub const Yi: Script = Script(41);
     pub const ZanabazarSquare: Script = Script(177);
 }
+
+/// Enumerated property Grapheme_Cluster_Break.
+///
+/// See "Default Grapheme Cluster Boundary Specification" in UAX #29 for the
+/// summary of each property value:
+/// <https://www.unicode.org/reports/tr29/#Default_Grapheme_Cluster_Table>
+///
+/// The numeric value is compatible with `UGraphemeClusterBreak` in ICU4C.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(transparent)]
+pub struct GraphemeClusterBreak(pub u8);
+
+#[allow(missing_docs)] // These constants don't need individual documentation.
+#[allow(non_upper_case_globals)]
+impl GraphemeClusterBreak {
+    pub const Other: GraphemeClusterBreak = GraphemeClusterBreak(0); // name="XX"
+    pub const Control: GraphemeClusterBreak = GraphemeClusterBreak(1); // name="CN"
+    pub const CR: GraphemeClusterBreak = GraphemeClusterBreak(2); // name="CR"
+    pub const Extend: GraphemeClusterBreak = GraphemeClusterBreak(3); // name="EX"
+    pub const L: GraphemeClusterBreak = GraphemeClusterBreak(4); // name="L"
+    pub const LF: GraphemeClusterBreak = GraphemeClusterBreak(5); // name="LF"
+    pub const LV: GraphemeClusterBreak = GraphemeClusterBreak(6); // name="LV"
+    pub const LVT: GraphemeClusterBreak = GraphemeClusterBreak(7); // name="LVT"
+    pub const T: GraphemeClusterBreak = GraphemeClusterBreak(8); // name="T"
+    pub const V: GraphemeClusterBreak = GraphemeClusterBreak(9); // name="V"
+    pub const SpacingMark: GraphemeClusterBreak = GraphemeClusterBreak(10); // name="SM"
+    pub const Prepend: GraphemeClusterBreak = GraphemeClusterBreak(11); // name="PP"
+    pub const RegionalIndicator: GraphemeClusterBreak = GraphemeClusterBreak(12); // name="RI"
+    pub const ZWJ: GraphemeClusterBreak = GraphemeClusterBreak(17); // name="ZWJ"
```

Comment:
Good question. ICU of course has constants for these values because at some point some characters had them, the constants are stable API, and we also have API that lets people map between value names and enum constant values.

You don't need to support data from old Unicode versions, so maybe you don't need now-unused values.

However, another question is how `\p{gcb=E_Base}` would be supported. It should yield an empty set rather than an unknown-value error. Is it easier to do so if unused values are defined?

So... my hunch is that it's cleaner / more user-friendly to define all values, even ones that currently aren't used.

Reply:
Question: is this set of 18 values stable, or will it grow over time?

If the set is stable, I think we should use a numeric enum rather than a newtype.  (If the set will grow, I think the newtype is correct.)

Reply:
Most enumerated properties can get new values added in future versions of Unicode. The ...Break properties frequently do.